src/super_gradients/training/dataloaders/dataloader_factory.py

```diff
@@ -0,0 +1,104 @@
+from hydra import initialize, compose
+import hydra
+from hydra.core.global_hydra import GlobalHydra
+from typing import Dict
+import super_gradients
+from torch.utils.data import BatchSampler, DataLoader
+from super_gradients.training.utils import get_param
+from super_gradients.training.datasets.detection_datasets import COCODetectionDataset
+from super_gradients.common.factories.samplers_factory import SamplersFactory
+from super_gradients.training.utils.distributed_training_utils import wait_for_the_master, get_local_rank
+from super_gradients.common.abstractions.abstract_logger import get_logger
+
+logger = get_logger(__name__)
+
+
+def get_data_loader(config_name, dataset_cls, train, dataset_params={}, dataloader_params={}):
+    """
+    Class for creating dataloaders for taking defaults from yaml files in src/super_gradients/recipes.
+
+    :param config_name: yaml config filename in recipes (for example coco2017_yolox).
+    :param dataset_cls: torch dataset uninitialized class.
+    :param train: controls whether to take
+        cfg.dataset_params.train_dataloader_params or cfg.dataset_params.valid_dataloader_params as defaults for the dataset constructor
+     and
+        cfg.dataset_params.train_dataset_params or cfg.dataset_params.valid_dataset_params as defaults for DataLoader contructor.
+
+    :param dataset_params: dataset params that override the yaml configured defaults, then passed to the dataset_cls.__init__.
+    :param dataloader_params: DataLoader params that override the yaml configured defaults, then passed to the DataLoader.__init__
+    :return: DataLoader
+    """
+    GlobalHydra.instance().clear()
+    with initialize(config_path="../../recipes"):
```

Comment:
use pkg_resources.resource_filename

Reply:
not possible here, Hydra expects relative path.

Reply:
will it work when installing from pip?

Reply:
Added hydra initialisation with absolute path in https://github.com/Deci-AI/super-gradients/pull/328/commits/0017eedb1ad94dbb6e06bb3c254c6c1eaf9a9ea3